tools/hive-bigquery/hive_component.py

```diff
@@ -0,0 +1,1118 @@
+                                    hive_table_model.tracking_table_name,
+                                    identifier, table_name, inc_col_min,
+                                    inc_col_max, clause, file_path)
+                    else:
+                        query = "INSERT INTO {0} (table_name,clause," \
+                                "file_path,gcs_copy_status,bq_job_id," \
+                                "bq_job_retries,bq_job_status) VALUES('{1}'," \
+                                "'{2}','{3}','TODO','TODO',0,'TODO')".format(
+                                    hive_table_model.tracking_table_name,
+                                    table_name, clause, file_path)
+                    # Commits information about the staging files
+                    mysql_component.execute_transaction(query)
+
+                query = "DELETE FROM {0} WHERE table_name='{1}' AND clause " \
+                        "='{2}' AND file_path='TODO'".format(
+                            hive_table_model.tracking_table_name,
+                            table_name, clause)
+                mysql_component.execute_transaction(query)
+                # Copies files from HDFS to GCS
+                gcs_component.stage_to_gcs(mysql_component, bq_component,
+                                           hive_table_model, bq_table_model,
+                                           gcs_bucket_name)
+
+    @staticmethod
+    def compare_max_values(hive_table_model, old_max, new_max):
+        """Compares the previously obtained maximum value with the newly
+        obtained maximum value of incremental column
+
+        Args:
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            old_max (str): Maximum value of the incremental column from the
+                tracking table
+            new_max (str): Maximum value of the incremental column from the
+                Hive table
+
+        Returns:
+            boolean: True if new_max value is greater than the old_max value,
+            else False
+        """
+
+        if hive_table_model.inc_col_type == "ts":
+            old_max = str(old_max)
+            new_max = str(new_max)
+
+            try:
+                old_max = datetime.datetime.strptime(old_max,
+                                                     "%Y-%m-%d %H:%M:%S.%f")
+                new_max = datetime.datetime.strptime(new_max,
+                                                     "%Y-%m-%d %H:%M:%S.%f")
+            except ValueError:
+                try:
+                    old_max = datetime.datetime.strptime(old_max,
+                                                         "%Y-%m-%d %H:%M:%S")
+                    new_max = datetime.datetime.strptime(new_max,
+                                                         "%Y-%m-%d %H:%M:%S")
+                except ValueError:
+                    old_max = datetime.datetime.strptime(old_max, "%Y-%m-%d")
```

Comment:
Date comparison is relying on string comparison and then there is type cast to int. Can we use date modules to apply the same logic that is being done?